lms/services/lti_outcomes.py

```diff
@@ -191,34 +191,38 @@ def _send_request(outcomes_request_params, pox_body):
         force_include_body=True,
     )
 
-    # Execute request and check HTTP status.
     try:
         response = requests.post(
             url=outcomes_request_params.lis_outcome_service_url,
             data=xml_body,
             headers={"Content-Type": "application/xml"},
             auth=oauth_client,
         )
+        # The following will raise ``requests.exceptions.HTTPError`` if
+        # there was an HTTP-related problem with the request. This exception
+        # is a subclass of ``requests.exceptions.RequestError``.
         response.raise_for_status()
     except RequestException as err:
-        raise ExternalRequestError(
+        # Handle any kind of ``RequestException``, be it an ``HTTPError`` or other
+        # flavor of ``RequestException``.
+        raise LTIOutcomesAPIError(
             "Error calling LTI Outcomes service", response
         ) from err
 
     # Parse response and check status code embedded in XML.
     try:
         xml = ElementTree.fromstring(response.text)
     except ElementTree.ParseError as err:
-        raise ExternalRequestError(
+        raise LTIOutcomesAPIError(
             "Unable to parse XML response from LTI Outcomes service", response
         ) from err
 
     status = find_element(xml, ["imsx_statusInfo", "imsx_codeMajor"])
     if status is None:
-        raise ServiceError("Failed to read status from LTI outcome response")
+        raise LTIOutcomesAPIError("Failed to read status from LTI outcome response")
 
     if status.text != "success":
-        raise ServiceError("LTI outcome request failed")
+        raise LTIOutcomesAPIError("LTI outcome request failed")
```

Comment:
It is possible to provide a more nuanced error response body by using the text value of the Element returned by `find_element(xml, ["imsx_statusInfo", "imsx_description"])` and using that text as the `details` argument to the `LTIOutcomesAPIError` constructor . However, the complexity of the tests for this service module were making my eyes cross, so I am not including it in this body of work.

Reply:
Ugh, yeah those tests look problematic, sorry

Reply:
I'm not a fan of the tests themselves either, but I think part of the problem here is that `LTIOutcomesClient` contains both logic and XML response handling, the class is too complex and that's most evident in its tests being too complex. The code for parsing XML and extracting certain values from XML belongs in a response validation schema.

And code for using templates for format XML strings belongs in ... well, templates.

OAuth 1 request signing might be separable too.

And calling the requests library and handling its responses and exceptions in a particular way.

Just a lot of separable and potentially reusable things run together in this one module, I think